prow/cmd/gangway/main.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/test-infra/pkg/flagutil"
+	prowcr "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowv1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	configflagutil "k8s.io/test-infra/prow/flagutil/config"
+	"k8s.io/test-infra/prow/gangway"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	client         prowflagutil.KubernetesOptions
+	github         prowflagutil.GitHubOptions
+	port           int
+	cookiefilePath string
+
+	config configflagutil.ConfigOptions
+
+	dryRun                 bool
+	gracePeriod            time.Duration
+	instrumentationOptions prowflagutil.InstrumentationOptions
```

Comment:
nit: the clusters look a bit messy, I don't understand why exactly these three groups were clustered together

Reply:
I don't understand either. I just removed the spaces to make the formatter treat it as 1 group instead of 3, which appears to have helped out gofmt.